Source/Core/Editor/UI/Drawers/ProcessSceneReferenceDrawer.cs

```diff
@@ -122,28 +122,55 @@ private void DrawDragAndDropArea(ref Rect rect, Action<object> changeValueCallba
             DropAreaGUI(ref rect, ref guiLineRect, reference, droppedGameObject, changeValueCallback, label);
         }
 
-        private void DrawMisconfigurationOnSelectedGameObjects(ProcessSceneReferenceBase reference, Type valueType, ref Rect originalRect, ref Rect guiLineRect)
+        private void HandleUnconfiguredObjects(ProcessSceneReferenceBase reference, Type valueType, ref Rect originalRect, ref Rect guiLineRect)
         {
-
             // Find all GameObjects that are missing the the component "valueType" needed
             IEnumerable<GameObject> gameObjectsWithMissingConfiguration = reference.Guids
                 .SelectMany(guidToDisplay => RuntimeConfigurator.Configuration.SceneObjectRegistry.GetObjects(guidToDisplay))
                 .Select(sceneObject => sceneObject.GameObject)
                 .Where(sceneObject => sceneObject == null || sceneObject.GetComponent(valueType) == null)
                 .Distinct();
 
-
-            // Add FixIt all if more than one game object exist
-            if (gameObjectsWithMissingConfiguration.Count() > 1)
+            // Automatically fix objects if configuration says so
+            if (AdvancedSettings.Instance.AutoAddProperties && gameObjectsWithMissingConfiguration.Count() > 0)
             {
-                guiLineRect = AddNewRectLine(ref originalRect, EditorDrawingHelper.SingleLineHeight);
-                AddFixItAllButton(gameObjectsWithMissingConfiguration, valueType, ref originalRect, ref guiLineRect);
-            }
+                Dictionary<GameObject, Component[]> alreadyAttachedProperties = new Dictionary<GameObject, Component[]>();
+                foreach (GameObject gameObject in gameObjectsWithMissingConfiguration)
+                {
+                    alreadyAttachedProperties.Add(gameObject, gameObject.GetComponents(typeof(Component)));
+                }
 
-            // Add FixIt on each component
-            foreach (GameObject selectedGameObject in gameObjectsWithMissingConfiguration)
+                RevertableChangesHandler.Do(new ProcessCommand(
+                    () =>
+                    {
+                        foreach (GameObject gameObject in gameObjectsWithMissingConfiguration)
+                        {
+                            SceneObjectAutomaticSetup(gameObject, valueType);
+                        }
+                    },
+                    () =>
+                    {
+                        foreach (GameObject gameObject in gameObjectsWithMissingConfiguration)
+                        {
+                            UndoSceneObjectAutomaticSetup(gameObject, valueType, alreadyAttachedProperties[gameObject]);
+                        }
+                    }
+                   ));
+            }
+            else
             {
```

Comment:
I think this is okay to do but we need to write in the settings that setting the hook will automatically "click" all fxitit buttons.

Reply:
What do you mean? We have "Automatically add required properties when referencing a scene object in the Step Inspector". Why mention the fix it buttons?

Reply:
Yes you are correct, it will add properties to any reference with missing properties that is shown in the inspector. Not every missing property in the process, but it's definitely worth adding a note.